examples/tv-app/tv-common/clusters/messages/MessagesManager.h

```diff
@@ -21,20 +21,116 @@
 
 #include <iostream>
 #include <list>
+#include <vector>
+
+struct CachedMessageOption
+{
+    CachedMessageOption(uint32_t id, std::string label) :
+        mLabel(label), mOption{ chip::MakeOptional(id), chip::MakeOptional(chip::CharSpan::fromCharString(mLabel.c_str())) }
+    {}
+
+    CachedMessageOption(const CachedMessageOption & option) :
+        mLabel(option.mLabel),
+        mOption{ option.mOption.messageResponseID, chip::MakeOptional(chip::CharSpan::fromCharString(mLabel.c_str())) }
+    {}
+
+    // CachedMessageOption & operator=(const CachedMessageOption & option) { return *this; };
+
+    chip::app::Clusters::Messages::MessageResponseOption GetMessageOption() { return mOption; }
+
+    ~CachedMessageOption() {}
+
+protected:
+    std::string mLabel;
+    chip::app::Clusters::Messages::MessageResponseOption mOption;
+};
+
+struct CachedMessage
+{
+    chip::ByteSpan mMessageId;
+
+    CachedMessage(const CachedMessage & message) :
+        mPriority(message.mPriority), mMessageControl(message.mMessageControl), mStartTime(message.mStartTime),
+        mDuration(message.mDuration), mMessageText(message.mMessageText), mOptions(message.mOptions)
+    {
+        memcpy(mMessageIdBuffer, message.mMessageIdBuffer, sizeof(mMessageIdBuffer));
+        mMessageId = chip::ByteSpan(mMessageIdBuffer);
+
+        for (CachedMessageOption & entry : mOptions)
+        {
+            mResponseOptions.push_back(entry.GetMessageOption());
+        }
+    }
+
+    // CachedMessage & operator=(const CachedMessage & message) { return *this; };
```

Comment:
same here: you have to either =delete or implement. You cannot leave it as is because this structure contains self-referencing pointers and a compiler-generated bitwise assignment will fail.

Reply:
marked the assignment operator as delete